packages/-ember-data/tests/integration/adapter/rest-adapter-test.js

```diff
@@ -998,6 +998,82 @@ module('integration/adapter/rest_adapter - REST Adapter', function (hooks) {
       });
   });
 
+  test('updateRecord - hasMany relationships faithfully reflect removal from response', async function (assert) {
+    Post.reopen({ comments: DS.hasMany('comment', { async: false }) });
+    Comment.reopen({ post: DS.belongsTo('post', { async: false }) });
```

Comment:
we should register a fresh class here vs using reopen (we've RFC'd deprecating reopen and need to cleanup our own tests). Same for the test below.

Reply:
ok, like [that](https://github.com/emberjs/data/pull/7512/commits/9080033d14b6166f19d17fdc3b3ab147dce9842e)?

Reply:
exactly, thanks!